sql/plan/exchange.go

```diff
@@ -97,34 +99,40 @@ func newExchangeRowIter(
 		started:     false,
 		tree:        tree,
 		partitions:  iter,
-		quit:        make(chan struct{}),
+		quitChan:    make(chan struct{}),
 	}
 }
 
 func (it *exchangeRowIter) releaseToken() {
 	it.mut.Lock()
 	defer it.mut.Unlock()
 
-	if it.tokens != nil {
-		it.tokens <- struct{}{}
+	if it.tokensChan != nil {
+		it.tokensChan <- struct{}{}
 	}
 }
 
 func (it *exchangeRowIter) closeTokens() {
 	it.mut.Lock()
 	defer it.mut.Unlock()
 
-	close(it.tokens)
-	it.tokens = nil
+	close(it.tokensChan)
+	it.tokensChan = nil
+}
+
+func (it *exchangeRowIter) tokens() chan struct{} {
+	it.mut.RLock()
```

Comment:
I don't get this, won't this lock-unlock with nothing done in the middle? Or does this cover the case of `it.tokensChan` being closed just in the middle of the return?

Reply:
note the `defer` before the `RUnlock`. Basically, this is the same as:

```go
it.mut.RLock()
result := it.tokensChan
it.mut.RUnlock()
return result
```